src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -239,10 +239,32 @@ public override async Task<CompletionChange> GetChangeAsync(Document document, C
                 assemblyName += $", PublicKey={ publicKey }";
             }
 
-            var textChange = new TextChange(item.Span, assemblyName);
+            var span = await GetTextChangeSpanAsync(document, item, cancellationToken);
+            var textChange = new TextChange(span, assemblyName);
             return CompletionChange.Create(textChange);
         }
 
+        private static async Task<TextSpan> GetTextChangeSpanAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var span = item.Span;
+            var syntaxFactsService = document.GetLanguageService<ISyntaxFactsService>();
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(item.Span.Start);
+            if (syntaxFactsService.IsStringLiteral(token) && !string.IsNullOrEmpty(token.ValueText))
+            {
+                span = token.Span;
+                var positionOfValueIntText = token.Text.IndexOf(token.ValueText);
+                if (positionOfValueIntText >= 0)
+                {
+                    span = new TextSpan(
+                        start: token.Span.Start + positionOfValueIntText,
+                        length: token.ValueText.Length);
```

Comment:
This approach is horrible, but I couldn't figure out how to make sure the completion item spans the whole string while preserving any existing double quotes. @CyrusNajmabadi you probably know a better way how to fix this.

Reply:
There are two approaches you can take here.  I'm fine with either:

1. Walk backwards until you hit the quote character.   Not great, but workable.  Would be incorrect if there was a `\"` in the string.
2. you have the start of the string.  and you can find out if it's regular or varbatim.  if it's regular, then the position you want is `start + "\"".Length`.  If it's verbatim you want `start + "@\"".Length`.

I'd prefer approach #2.